include/earcut.hpp

```diff
@@ -89,7 +84,43 @@ class Earcut {
     double minY, maxY;
     double size;
 
-    std::unique_ptr<boost::object_pool<Node>> nodes;
+    template <typename T, typename Alloc = std::allocator<T>>
+    class ObjectPool {
+    public:
+        ObjectPool() { }
+        ObjectPool(std::size_t blockSize) {
+            reset(blockSize);
+        }
+        ~ObjectPool() {
+            clear();
+        }
+        template <typename... Args>
+        T* construct(Args&&... args) {
+            if (currentIndex >= blockSize) {
+                currentBlock = alloc.allocate(blockSize);
+                allocations.emplace_back(currentBlock);
+                currentIndex = 0;
+            }
+            T* object = &currentBlock[currentIndex++];
+            alloc.construct(object, std::forward<Args>(args)...);
+            return object;
+        }
+        void reset(std::size_t newBlockSize) {
+            currentBlock = nullptr;
+            currentIndex = -1;
+            for (auto allocation : allocations) alloc.deallocate(allocation, blockSize);
+            allocations.clear();
+            blockSize = (std::max<std::size_t>)(1, newBlockSize);
+        }
+        void clear() { reset(blockSize); }
+    private:
+        T* currentBlock = nullptr;
+        std::size_t currentIndex = -1;
```

Comment:
(std::numeric_limits<std::size_t>max)() would work, but it's pretty long to write, and -1 does the job. I can change it if you insist.